src/lib/fileutils.ts

```diff
@@ -11,31 +11,55 @@ import {
   IUser
 } from "../types";
 
+export const generateHldLifecyclePipelineYaml = async (projectRoot: string) => {
+  logger.info(
+    `Generating hld lifecycle pipeline hld-lifecycle.yaml in ${projectRoot}`
+  );
+
+  const azurePipelinesYamlPath = path.join(projectRoot, "hld-lifecycle.yaml");
+
+  if (fs.existsSync(azurePipelinesYamlPath)) {
+    logger.warn(
+      `Existing hld-lifecycle.yaml found at ${azurePipelinesYamlPath}, skipping generation`
+    );
+
+    return;
+  }
+
+  const hldYaml = hldLifecyclePipelineYaml();
+  logger.info(`Writing hld-lifecycle.yaml file to ${azurePipelinesYamlPath}`);
+  fs.writeFileSync(azurePipelinesYamlPath, hldYaml, "utf8");
+};
+
 /**
  * Writes out the starter azure-pipelines.yaml file to `targetPath`
  *
  * @param targetPath Path to write the azure-pipelines.yaml file to
```

Comment:
Update documentation on params.

Also why would we need the filename to be variable?

Reply:
Followup, should we rename the pipelines to be more descriptive of what each one entails and should we also rename the functions for starterAzurePipelines to be reflect which pipeline it's for. @evanlouie @bnookala